cpp/include/raft/linalg/rsvd.cuh

```diff
@@ -18,13 +18,94 @@
 
 #pragma once
 
-#include "detail/rsvd.cuh"
-
 #include <raft/core/device_mdspan.hpp>
+#include <raft/linalg/detail/rsvd.cuh>
 
 namespace raft {
 namespace linalg {
 
+/**
+ * @brief randomized singular value decomposition (RSVD)
+ * @tparam math_t the data type
+ * @tparam idx_t index type
+ * @param[in]  handle:  raft handle
+ * @param[in]  in:      input matrix in col-major format. On exit this matrix is destroyed
+ *                      [dim = n_rows * n_cols]
+ * @param[out] S:       array of singular values of input matrix.
+ *                      [dim = k]
+ * @param[out] U:       optional left singular values of input matrix. Use std::nullopt to not
+ * generate it. [dim = n_rows * k]
+ * @param[out] V:       optional right singular values of input matrix. Use std::nullopt to not
+ * generate it. [dim = k * n_cols]
+ * @param[in]  k:       Rank of the k-SVD decomposition of matrix in. Number of singular values to
+ * be computed. The rank is less than min(m,n).
```

Comment:
I think this should say the rank must be less than or equal to min(m, n).

Reply:
According to cuSolver doc:
`Rank of the k-SVD decomposition of matrix A. rank is less than min(m,n). `

Reply:
My point here is that this should read "rank must be..." rather than "rank is". In practice, it doesn't make much sense for the rank to be the number of columns in the original data set anyways, but we should be letting the user know what the expected rank "should" be,not what it "is". Does that make sense?

Reply:
Yes thanks for the clarification!